inst/templates/ad_adtpet.R

```diff
@@ -54,28 +54,29 @@ attr(param_lookup$NVTESTCD, "label") <- "NV Test Short Name"
 # Get list of ADSL vars required for derivations
 adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01A, TRT01P)
 
-adpet <- nv %>%
+adtpet <- nv %>%
   ## Join ADSL with NV (need TRTSDT for ADY derivation) ----
   derive_vars_merged(
     dataset_add = adsl,
     new_vars = adsl_vars,
     by_vars = get_admiral_option("subject_keys")
   ) %>%
-  ## Join ADPET with AG for tracer information ----
+  ## Join ADTPET with AG for tracer information ----
   # Users can add more variables in the `new_vars` argument as needed.
   derive_vars_merged(
     dataset_add = ag,
     new_vars = exprs(AGTRT, AGCAT),
-    by_vars = exprs(USUBJID, VISIT, NVLNKID = AGLNKID)
+    by_vars = c(get_admiral_option("subject_keys"), exprs(VISIT, NVLNKID = AGLNKID))
   ) %>%
```

Comment:
I think it is better to add the filter immediately after AG is merged so we don't need to add additional columns such as ADT and ADY to uneeded rows, 
filter(AGCAT == "TAU TRACER") # Filter nv dataset for tau records only